### PR TITLE
Add mono recipe

### PR DESCRIPTION
**Pull request by bgruening**
This will add a the mono runtime environment.

### CODE REVIEW

**Comment by bgruening** [comment]
@jakirkham 
I think we need help here. This does not look like an recipe error.

```
Traceback (most recent call last):
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/builder.py", line 200, in build
    conda_build.api.build(meta.meta, config=config)
AttributeError: module 'conda_build' has no attribute 'api'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/bin/conda-build-all", line 11, in <module>
    load_entry_point('conda-build-all==1.0.0', 'console_scripts', 'conda-build-all')()
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/cli.py", line 90, in main
    b.main()
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/builder.py", line 263, in main
    built_dist_location = self.build(meta, build_config)
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/builder.py", line 204, in build
    return bldpkg_path(build.build(meta.meta))
  File "/opt/conda/lib/python3.5/site-packages/conda_build_all/build.py", line 34, in build
    build_module.build(meta, post=None, need_source_download=True, **kwd)
  File "/opt/conda/lib/python3.5/site-packages/conda_build/build.py", line 650, in build
    preserve_egg_dir=bool(m.get_value('build/preserve_egg_dir')))
  File "/opt/conda/lib/python3.5/site-packages/conda_build/post.py", line 194, in post_process
    compile_missing_pyc(files)
  File "/opt/conda/lib/python3.5/site-packages/conda_build/post.py", line 189, in compile_missing_pyc
    call([python_exe, '-Wi', '-m', 'py_compile', f], cwd=cwd)
  File "/opt/conda/lib/python3.5/subprocess.py", line 557, in call
    with Popen(*popenargs, **kwargs) as p:
  File "/opt/conda/lib/python3.5/subprocess.py", line 947, in __init__
    restore_signals, start_new_session)
  File "/opt/conda/lib/python3.5/subprocess.py", line 1551, in _execute_child
    raise child_exception_type(errno_num, err_msg)
FileNotFoundError: [Errno 2] No such file or directory: '/opt/conda/envs/_build_placehold_placehold_placehold_placehold_placehold_placeho/bin/python'
```

File: recipes/mono/bld.bat [hunk 1]
```diff
@@ -0,0 +1,3 @@
+./configure --prefix=$PREFIX --host=i686-w64-mingw32
```
Comment (bgruening) [inline recipes/mono/bld.bat]: Thanks, I skiped th build.

Comment (bgruening) [inline recipes/mono/bld.bat]: But obviously the original error is the same.

**Comment by bgruening** [comment]
@jakirkham can we merge this?
Thanks @blankclemens!

File: recipes/mono/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [not linux]
+
+requirements:
+  build:
+    - python
+    - gcc       # [unix]
```
Comment (jakirkham) [inline recipes/mono/meta.yaml]: Why do we need `gcc`? Are the system compilers not sufficient?

**Comment by bgruening** [comment]
@jakirkham removed it! Thanks!

File: recipes/mono/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  fn: {{ name }}-{{ version }}.tar.bz2
+  url: http://download.mono-project.com/sources/{{ name }}/{{ name }}-{{ version }}.tar.bz2
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  skip: True  # [not linux]
+
+requirements:
+  build:
```
Comment (jakirkham) [inline recipes/mono/meta.yaml]: Also please add `toolchain`.

File: recipes/mono/build.sh [hunk 4]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+./configure --prefix=$PREFIX --with-glib=embedded --enable-nls=no
+
+make
```
Comment (jakirkham) [inline recipes/mono/build.sh]: Is there a `make check` that we can run?

**Comment by jakirkham** [comment]
Also please add `[skip appveyor]` in commit messages if not building on Windows.

**Comment by jakirkham** [comment]
Appears there is only one test that fails, `delegate2.exe`. Also it looks like upstream [disabled]( https://github.com/mono/mono/blame/a1ac272ffe8d6b54bc9f9ebe1331699d1b623507/mono/tests/Makefile.am#L603-L604 ) it too. Maybe we can do the same?

**Comment by bgruening** [comment]
@jakirkham the change was made 3 years ago, Oo. I tried to exclude this tests from the command line but did not succeed so far any idea? Can we remove make check?

**Comment by jakirkham** [comment]
Indeed. Weird that it still runs that test anyways. :confused:

Sure. Comment it out. Make a note to uncomment it later. Also raise an issue in the feedstock.

**Comment by bgruening** [comment]
Arg sorry, I messed up the skip CI step :(
Making an issue in the feedstock is only possible after merge, correct? I will do this then!

Thanks!

**Comment by bgruening** [comment]
@jakirkham thanks a lot!

**Pull request merged**